snapcraft/internal/sources.py

```
@@ -389,7 +389,13 @@ def get(sourcedir, builddir, options):
     handler_class = _get_source_handler(source_type, options.source)
     handler = handler_class(options.source, sourcedir, source_tag,
                             source_branch)
-    handler.pull()
+    try:
+        handler.pull()
+    except FileNotFoundError:
```

Though, this might mask errors for zip, tar, and local source types.